batch.go

```diff
@@ -77,17 +99,28 @@ func (ebi *EventBatchIterator) Next(eventID string, opts *BatchOptions) (*EventB
 		}
 	}
 
+	var key string
+	for partitionKey, visit := range ebi.Visited {
+		if !visit {
+			key = partitionKey
+			ebi.Visited[partitionKey] = true
```

Comment:
What if `Next` is called and `sizeof(ebi.PartitionEventsMap[key]) > MaxEventSize`? That is to say, the size of the batch message for all of the events in a partition is greater than the max size of an Event Hub message.

I think this logic will mark the partition as visited even if there are events still left after the first batch is built and returned. I believe you want to keep a pointer to the last visited event in a partition and keep consuming a partition until you reach the end of the event slice.

Reply:
done